### DISCUSSION TITLE
Add mechanism to control `addPlugin` order

### DISCUSSION THREAD

**Discussion by zachleat** [11ty/eleventy#4069] [Enhancement Queue]
Need to make sure https://github.com/11ty/eleventy-upgrade-help runs last. Internally, Eleventy uses `defaultConfig.js` to ensure HTML transformer plugin is added first and runs first.

Might be as simple as a `priority` option in the second argument to `addPlugin` that we use to sort the plugin array.

cc @cfjedimaster

Related https://github.com/11ty/eleventy-upgrade-help/issues/8

**Comment by zachleat** [comment]
> By default this repo closes enhancement request issues and places them in a queue for folks to upvote. We do this to avoid a large build-up of open issues. We got this idea from the [lodash project](https://twitter.com/samselikoff/status/991395669016436736).

> Don’t forget to upvote the first comment with 👍 to register your vote! You can view [the current enhancement backlog](https://github.com/11ty/eleventy/issues?utf8=%E2%9C%93&q=label%3Aneeds-votes+sort%3Areactions-%2B1-desc+).

**Comment by cfjedimaster** [comment]
If there have been other plugins in the past that required last, or first, then a priority option would be awesome. Of course, plugins would have to contend with 2 or more wanting the first/last position.

**Comment by zachleat** [comment]
Want to also cross link the `immediate` option here too: https://www.11ty.dev/docs/plugins/#advanced-execute-a-plugin-immediately
